pkg/nsx/services/realizestate/realize_state.go

```diff
@@ -62,9 +62,6 @@ func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, inte
 		}
 		entitiesRealized := 0
 		for _, result := range results.Results {
-			if entityType != "" && result.EntityType != nil && *result.EntityType != entityType {
-				continue
-			}
 			if *result.State == model.GenericPolicyRealizedResource_STATE_REALIZED {
```

Comment:
I prefer to do it like this with function. No need to explicitly compare the count of model.GenericPolicyRealizedResource_STATE_REALIZED resources. Instead, I assume the intent of this function is all NSX returned resources should be with `model.GenericPolicyRealizedResource_STATE_REALIZED` , then we can directly return error if any is not in this state. 

```
func() error {
		results, err := service.NSXClient.RealizedEntitiesClient.List(intentPath, nil)
		err = nsxutil.TransNSXApiError(err)
		if err != nil {
			return err
		}
		for _, result := range results.Results {
			if *result.State == model.GenericPolicyRealizedResource_STATE_ERROR {
				var errMsg []string
				for _, alarm := range result.Alarms {
					if alarm.Message != nil {
						errMsg = append(errMsg, *alarm.Message)
					}
				}
				return NewRealizeStateError(fmt.Sprintf("%s realized with errors: %s", *result.EntityType, errMsg))
			}
			if *result.State != model.GenericPolicyRealizedResource_STATE_REALIZED {
				return fmt.Errorf("%s not realized", entityType)
			}
		}
		return nil
	}
```

Reply:
It seems there is a problem.
If you have 3 items. One is unrealized, one is error, one is realized.
Your logic depends on the return order.